src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -44,21 +51,45 @@ const EditShiftHours = ({
 
   const inputName = 'shift';
   const { setServiceError, userId } = useApplicationContext();
+  const [isChecked, setIsChecked] = useState(false);
+  const [redirectTarget, setRedirectTarget] = useState('');
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [clashingProperty, setClashingProperty] = useState(null);
+  const [clashingTimes, setClashingTimes] = useState(null);
+  const navigate = useNavigate();
+
   const {
     timeEntries,
     setTimeEntries,
     timecardDate,
     summaryErrors,
     setSummaryErrors,
+    setIsAlertVisible,
+    summaryMessages,
+    setSummaryMessages,
   } = useTimecardContext();
 
-  const [clashingProperty, setClashingProperty] = useState(null);
-  const [clashingTimes, setClashingTimes] = useState(null);
-
   useEffect(() => {
     focusErrors(document.getElementById('summary-error-0-message'));
+    if (shouldRedirect) {
+      navigate(redirectTarget);
+    }
+
+    if (summaryMessages && Object.keys(summaryMessages).length !== 0) {
+      setTimeout(() => setIsAlertVisible(false), 10000);
+    }
   }, [summaryErrors]);
 
+  const handleChange = () => {
```

Comment:
This name suggests a generic handler but this is specifically for the checkbox.
We should update the name or place the arrow function directly in the handler param of the checkbox.

Reply:
Updated to be relevant to the checkbox only functionality